src/data_info/services/stat.clj

```diff
@@ -46,76 +65,107 @@
+    (assoc stat-map
+           :label (paths/path->label user path))
+    stat-map))
 
 
 (defn- merge-type-info
-  [stat-map cm user path]
-  (if-not (is-dir? stat-map)
+  [stat-map cm user path included-keys]
+  (if (and (needs-any-key? included-keys :infoType :content-type) (not (is-dir? stat-map)))
     (assoc stat-map
-      :infoType     (get-types cm user path)
-      :content-type (irods/detect-media-type cm path))
+      :infoType     (if (needs-key? included-keys :infoType) (get-types cm user path))
+      :content-type (if (needs-key? included-keys :content-type) (irods/detect-media-type cm path)))
     stat-map))
 
-
 (defn ^IPersistentMap decorate-stat
-  [^IPersistentMap cm ^String user ^IPersistentMap stat]
+  [^IPersistentMap cm ^String user ^IPersistentMap stat included-keys]
   (let [path (:path stat)]
     (-> stat
-      (assoc :id         (-> (meta/get-attribute cm path uuid/uuid-attr) first :value)
-             :permission (perm/permission-for cm user path))
-      (merge-label user path)
-      (merge-type-info cm user path)
-      (merge-shares cm user path)
-      (merge-counts cm user path))))
-
+      (assoc :id         (if (needs-key? included-keys :id) (-> (meta/get-attribute cm path uuid/uuid-attr) first :value))
+             :permission (if (needs-key? included-keys :permission) (perm/permission-for cm user path)))
+      (merge-label user path included-keys)
+      (merge-type-info cm user path included-keys)
+      (merge-shares cm user path included-keys)
+      (merge-counts cm user path included-keys)
+      (select-keys included-keys))))
+
+(defn- get-filter-set
+  [filter-vec-or-string default]
+  (if (nil? filter-vec-or-string)
+    (set default)
+    (if (string? filter-vec-or-string)
+      (set (map keyword (string/split filter-vec-or-string #",")))
+      (set (map keyword filter-vec-or-string)))))
+
+(defn- process-filters
+  "Process an include and an exclude string into just a list of keys to include"
+  [include exclude]
+  (let [all-keys #{:id :path :type :label :date-created :date-modified :permission :share-count :file-count :dir-count :file-size :content-type :infoType :md5}
+        includes-set (get-filter-set include all-keys)
+        excludes-set (get-filter-set exclude [])]
+      (cset/intersection all-keys (cset/difference includes-set excludes-set))))
 
 (defn ^IPersistentMap path-stat
-  [^IPersistentMap cm ^String user ^String path]
-  (let [path (ft/rm-last-slash path)]
+  [^IPersistentMap cm ^String user ^String path & {:keys [filter-include filter-exclude] :or {filter-include nil filter-exclude nil}}]
+  (let [path (ft/rm-last-slash path)
+        included-keys (process-filters filter-include filter-exclude)]
     (log/debug "[path-stat] user:" user "path:" path)
     (validators/path-exists cm path)
-    (decorate-stat cm user (info/stat cm path))))
+    (let [base-stat (if (needs-any-key? included-keys :type :date-created :date-modified :file-size :md5)
```

Comment:
The base stat also returns a `path` key and an `id` key (the latter is just set to the path though), but since we've already got a path if we're here, we don't need to fetch it, so it's not included in the list for checking.